mysys/my_getopt.c

```diff
@@ -969,7 +969,7 @@ my_bool getopt_compare_strings(register const char *s, register const char *t,
 /*
   function: eval_num_suffix
 
-  Transforms suffix like k/m/g to their real value.
+  Transforms suffix like k/m/g/t/p/e to their real value.
 */
 
 static inline long eval_num_suffix(char *suffix, int *error)
```

Comment:
long is not capable to hold values starting from T. Should be unsigned long long or similar data type.